independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -361,6 +373,27 @@ void init() {
         return bindings;
     }
 
+    static Map<DotName, ClassInfo> findTransitiveInterceptorBindings(Collection<DotName> initialBindings,
+            IndexView index, Map<DotName, ClassInfo> result) {
+        // any annotation that contains interceptor binding annotation is effectively also an interceptor binding
+        for (DotName annotationName : initialBindings) {
+            Set<DotName> collectedAnnotations = new HashSet<>();
+            for (AnnotationInstance annotation : index.getAnnotations(annotationName)) {
+                if (Kind.CLASS.equals(annotation.target().kind())
+                        && Modifier.isInterface(annotation.target().asClass().flags())) {
+                    // TODO is this good way to differentiate an annotation in Jandex?
```

Comment:
`Modifier.isInterface()` returns true even for normal interfaces, no? I don't think there is an easy way to differentiate annotations though - see https://github.com/wildfly/jandex/issues/60. I'm using this workaround: https://github.com/quarkusio/quarkus/blob/master/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/NoArgsConstructorProcessor.java#L151.

Reply:
Ah, I was wondering what was that magical trick for ;-)